app/controllers/hyrax/autopopulation/dashboard/work_fetchers_controller.rb

```diff
@@ -43,6 +98,23 @@ def ensure_authorized!
           def config_object
             Rails.application.config.hyrax_autopopulation
           end
+
+          # We are using array splat to pass multiple arguments to a method call
+          # the method below add an additional arguments depending on whether records are stored i redis or activerecord
+          #
+          # for regular argument the splat expects
+          # [autopopulation_settings_params]
+          #
+          # and for key word arguments it expetcts a hash
+          #   {user: current_user}
+          #
+          def pass_arguments_by_storage_type(value)
+            if config_object.storage_type == "activerecord"
```

Comment:
It would be good to move the "activerecord" and "redis" strings to a constant and refer to them there, rather than having strings scattered throughout the app.